src/components/policy/policy_external/test/sql_pt_representation_storage_test.cc

```diff
@@ -0,0 +1,121 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "gtest/gtest.h"
+#include "policy/sql_pt_representation.h"
+#include "utils/file_system.h"
+#include "policy/sql_wrapper.h"
+#include "policy/mock_policy_settings.h"
+#include "utils/shared_ptr.h"
+#include "utils/make_shared.h"
+
+#include <string>
+#include <memory>
+
+using testing::ReturnRef;
+using testing::NiceMock;
+
+namespace test {
+namespace components {
+namespace policy_test {
+
+class SQLPTRepresentationStorageTest : public ::testing::Test {
+ protected:
+  const std::string kAppStorageFolder =
+      "storage_SQLPTRepresentationStorageTest";
+
+  std::shared_ptr<policy::SQLPTRepresentation> reps;
+  std::shared_ptr<utils::dbms::SQLQuery> query_wrapper_;
+  std::shared_ptr<policy_handler_test::MockPolicySettings> policy_settings_;
+
+  void SetUp() {
+    file_system::CreateDirectory(kAppStorageFolder);
+
+    reps = std::make_shared<policy::SQLPTRepresentation>();
+    ASSERT_TRUE(reps != NULL);
+
+    policy_settings_ =
+        std::make_shared<NiceMock<policy_handler_test::MockPolicySettings> >();
+    ON_CALL(*policy_settings_, app_storage_folder())
+        .WillByDefault(ReturnRef(kAppStorageFolder));
+    EXPECT_EQ(::policy::SUCCESS, reps->Init(policy_settings_.get()));
+
+    query_wrapper_ = std::make_shared<utils::dbms::SQLQuery>(reps->db());
+    ASSERT_TRUE(query_wrapper_ != NULL);
+  }
+
+  void TearDown() OVERRIDE {
+    EXPECT_TRUE(reps->Drop());
+    EXPECT_TRUE(reps->Close());
+    reps->RemoveDB();
+
+    policy_settings_.reset();
```

Comment:
ZhdanovP why do you need this? All these objects will be destroyed automatically after teardown.